server/src/diagnostic_ext.rs

```diff
@@ -28,24 +29,33 @@ pub async fn publish_diagnostics_for_set(
         .filter_map(|uri| Some((convert_slice_url_to_uri(uri)?, vec![])))
         .collect::<HashMap<Url, Vec<tower_lsp::lsp_types::Diagnostic>>>();
 
-    // Process the diagnostics and populate the map
-    process_diagnostics(diagnostics, &mut map);
+    // Process the diagnostics and populate the map. Any diagnostics that do not have a span are returned for further processing.
```

Comment:
Where and for what processing?
> returned for further processing.

Reply:
Right below.

 Maybe saying further processing is misleading. It returns a `Vec` of the spanless diagnostics as it cannot do anything with them . We then take these and iterate through them. For each one we emit an error popup. This is what I am referring to as "processing".

Reply:
Yeah, I'd change the comment, since `process_diagnostics` theoretically does all the processing.
We just print them to a screen.